rpc/jsonrpc/client/http_json_client.go

```diff
@@ -277,12 +275,12 @@ func (c *Client) sendBatch(ctx context.Context, requests []*jsonRPCBufferedReque
 	return results, nil
 }
 
-func (c *Client) nextRequestID() rpctypes.JSONRPCIntID {
+func (c *Client) nextRequestID() int {
 	c.mtx.Lock()
+	defer c.mtx.Unlock()
 	id := c.nextReqID
 	c.nextReqID++
-	c.mtx.Unlock()
-	return rpctypes.JSONRPCIntID(id)
+	return id
```

Comment:
see comment below: but I worry about this for things like client restarts or clientID collisions.